pyfar/dsp/dsp_interpolation.py

```diff
@@ -0,0 +1,741 @@
+    window_stats : tuple
+        A tuple containing information about the smoothing process
+
+        `n_window`
+            The window length in (logarithmically spaced) samples
+        `window_width`
+            The actual window width in octaves. This can deviate due to the
+            ideal width because the window must have an integer length
+
+    Notes
+    -----
+    Method 3 in Tylka at al. 2017 is mathematically more elegant at the
+    price of a largely increased computational and memory cost. In most
+    practical cases, methods 2 and 3 yield close to identical results (cf. Fig.
+    2 and 3 in Tylka et al. 2017). If the spectrum contains extreme
+    discontinuities, however, method 3 is superior (see examples below).
+
+    References
+    ----------
+    .. [#] J. G. Tylka, B. B. Boren, and E. Y. Choueiri, “A Generalized Method
+           for Fractional-Octave Smoothing of Transfer Functions that Preserves
+           Log-Frequency Symmetry (Engineering Report),” J. Audio Eng. Soc. 65,
+           239-245 (2017). doi:10.17743/jaes.2016.0053
+
+    Examples
+    --------
+
+    Octave smoothing of continuous spectrum consisting of two bell filters.
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> signal = pf.signals.impulse(441)
+        >>> signal = pf.dsp.filter.bell(signal, 1e3, 12, 1, "III")
+        >>> signal = pf.dsp.filter.bell(signal, 10e3, -60, 100, "III")
+        >>> smoothed, _ = pf.dsp.smooth_fractional_octave(signal, 1)
+        >>> ax = pf.plot.freq(signal, label="input")
+        >>> pf.plot.freq(smoothed, label="smoothed")
+        >>> ax.legend(loc=3)
+
+    Octave smoothing of the discontinuous spectrum of a sine signal causes
+    artifacts at the edges due to the intermediate interpolation steps (cf.
+    Tylka et al. 2017, Fig. 4). However this is a rather unusual application
+    and is mentioned only for the sake of completeness.
+
+    .. plot::
+
+        >>> import pyfar as pf
+        >>> signal = pf.signals.sine(1e3, 4410)
+        >>> signal.fft_norm = "amplitude"
+        >>> smoothed, _ = pf.dsp.smooth_fractional_octave(signal, 1)
+        >>> ax = pf.plot.freq(signal, label="input")
+        >>> pf.plot.freq(smoothed, label="smoothed")
+        >>> ax.set_xlim(200, 4e3)
+        >>> ax.set_ylim(-45, 5)
+        >>> ax.legend(loc=3)
+    """
+
+    if not isinstance(signal, pf.Signal):
+        raise TypeError("Input signal has to be of type pyfar.Signal")
+
+    if mode in ["magnitude", "magnitude_copy"]:
+        data = [np.atleast_2d(np.abs(signal.freq_raw))]
+    elif mode == "complex":
+        data = [np.atleast_2d(np.real(signal.freq_raw)),
+                np.atleast_2d(np.imag(signal.freq_raw))]
+    elif mode == "magnitude_phase":
+        data = [np.atleast_2d(np.abs(signal.freq_raw)),
+                np.atleast_2d(pf.dsp.phase(signal, unwrap=True))]
+    else:
+        raise ValueError((f"mode is '{mode}' but must be 'magnitude', "
```

Comment:
One of the modes is lacking here

Reply:
Thanks for pointing out - the list is now complete